src/nix.rs

```diff
@@ -13,10 +13,11 @@ pub fn get() -> io::Result<OsString> {
     // host names are limited to `HOST_NAME_MAX` bytes
     //
     // https://pubs.opengroup.org/onlinepubs/9699919799/functions/gethostname.html
-    let size =
-        unsafe { libc::sysconf(libc::_SC_HOST_NAME_MAX) as libc::size_t };
+    let limit =
+        unsafe { libc::sysconf(libc::_SC_HOST_NAME_MAX) };
+    let size = if limit < 0 { 255 } else { limit } as usize;
 
-    let mut buffer = vec![0u8; size];
+    let mut buffer = vec![0u8; size + 1];
```

Comment:
Do you mind adding a comment stating that `+ 1` part is to ensure that the terminating null byte will also be included?

Reply:
I have added a comment as requested.